### DISCUSSION TITLE
A way to have error messages indicate the transformer step?

### DISCUSSION THREAD

**Discussion by flugruger** [nextgenhealthcare/connect#4851] [Q&A]
Is there a way to have Mirth errors indicate the originating transformer step number?
I'm getting errors like the one below. Since the code appears to be a concatenation of all the transformer steps, I've had to add comments indicating the step's name. 

Can Mirth indicate anything more specific than, "SCRIPT SOURCE: TRANSFORMER?"

------------------------
Transformer error
ERROR MESSAGE: Error evaluating transformer
com.mirth.connect.server.MirthJavascriptTransformerException: 
CHANNEL:	PracticeName ORM
CONNECTOR:	sourceConnector
SCRIPT SOURCE:	TRANSFORMER
SOURCE CODE:	
432: 	return true;
433: 
434: 	} // End of else, errors, API URL UPDATE
435: 
436: // SECOND, Put the value into the URL
437: var orderURL = configurationMap.get('WS_ORDER_URL').replace('LOCATION_ID', locID);
438: 
439: // ORDER NUMBER, UPDATE INTO URL (API URL UPDATE)
440: // Order Number from ORC-2-1, needed for URL
441: var orderNum = msg['ORC']['ORC.2']['ORC.2.1'].toString(); // Order Number
LINE NUMBER:	437
DETAILS:	Can't find method java.lang.String.replace(string,org.mozilla.javascript.Undefined).
	at d2a321dd-9b2f-45c2-a218-de38c4710097:437 (doTransform)

**Comment by pacmano1** [comment]
No.  However I think you will find over time you will one or two JS steps and not any of the other types so it becomes less painful.
